PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -101,16 +163,134 @@ def miniAOD_customizeCommon(process):
+                                                   ,'ak8PFJetsCHSSDSubJet:0JecFactor0', 'ak8PFJetsCHSSDSubJet:1JecFactor0'
+                                                   ,'ak8PFJetsCHSSDSubJet:0Flavour'   , 'ak8PFJetsCHSSDSubJet:1Flavour'
+                                                   ,'ak8PFJetsCHSSDSubJet:0CSVv2IVF'  , 'ak8PFJetsCHSSDSubJet:1CSVv2IVF']
 
+    ## Add CMS top tagger subjet info
+    process.ak8PFJetsCHSTopSubJet = cms.EDProducer("RecoJetDeltaRValueMapProducer",
+        src = cms.InputTag("ak8PFJetsCHS"),
+        matched = cms.InputTag("selectedPatJetsCMSTopTagPFJetsCHSPacked"),
+        distMax = cms.double(0.8),
+        values = cms.vstring(
+            "? numberOfDaughters > 0 ? daughterPtr(0).pt : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).pt : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).pt : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).pt : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).eta : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).eta : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).eta : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).eta : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).phi : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).phi : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).phi : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).phi : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).mass : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).mass : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).mass : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).mass : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).jecFactor(0) : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).jecFactor(0) : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).jecFactor(0) : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).jecFactor(0) : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).partonFlavour : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).partonFlavour : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).partonFlavour : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).partonFlavour : -9999",
+            "? numberOfDaughters > 0 ? daughterPtr(0).bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags') : -9999",
+            "? numberOfDaughters > 1 ? daughterPtr(1).bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags') : -9999",
+            "? numberOfDaughters > 2 ? daughterPtr(2).bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags') : -9999",
+            "? numberOfDaughters > 3 ? daughterPtr(3).bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags') : -9999",
```

Comment:
Indeed, 0 would probably be better from the compression point of view but in some cases could cause confusion since it would in the physically allowed range. However, with the CandidatePtr it would be easy to check if a subjet exists and if it doesn't, one would simply ignore any quantities.

Storing subjets as a collection of PackedCandidates should be possible but PackedCandidates cannot hold all relevant subjet information (they would be mostly useful for storing p4). So for the flavor, JEC factor, and btag discriminator, we would still have to store them as userFloats.